sw/device/lib/dif/dif_base.h

```diff
@@ -82,6 +84,50 @@ typedef enum dif_toggle {
   kDifToggleEnabled = 1,
 } dif_toggle_t;
 
+/**
+ * Checks if a DIF toggle type is valid.
+ *
+ * @param val A potential dif_toggle_t value.
+ * @return Bool indicating validity of toggle value.
+ */
+inline bool dif_is_valid_toggle(dif_toggle_t val) {
+  switch (val) {
+    case kDifToggleEnabled:
+      return true;
+    case kDifToggleDisabled:
+      return true;
+    default:
+      return false;
+  }
+}
+
+/**
+ * Converts a `dif_toggle_t` to a `bool`.
+ *
+ * @param val A dif_toggle_t value.
+ * @return Corresponding bool value.
+ */
+inline bool dif_toggle_to_bool(dif_toggle_t val) {
+  switch (val) {
+    case kDifToggleEnabled:
+      return true;
+    case kDifToggleDisabled:
+      return false;
```

Comment:
Thanks @timothytrippel for the PR.
Small nit: You could remove the line 115 as the `case kDifToggleDisabled` has the same return as the `default`

Reply:
Thanks for the review! @drewmacrae and @mcy requested I make this case explicit above (in the case the the dif_toggle_t ever changes in the future), so I think this may be best left as is (though you are correct it is a tad verbose).